src/dataflow/src/sink/kafka.rs

```diff
@@ -191,8 +201,17 @@ where
 
             let ret = move |input: &mut FrontieredInputHandle<_, _, _>| {
                 let producer: &RefCell<_> = producer.borrow();
-                let producer = &*producer.borrow();
-                let producer = match producer {
+                let producer_option = &mut *producer.borrow_mut();
```

Comment:
Have you been able to test this? This looks panicky to me, since you're taking out both an immutable and mutable borrow at once.

`return false;` will guarantee that the sink doesn't get scheduled again, right? If so you can reformulate like this, I think?

```rust
                if shutdown.load(Ordering::SeqCst) {
                    error!(
                        "encountered irrecoverable error. shutting down sink: {}",
                        name
                    );
                    return false;
                }

                let producer = producer.borrow();
                let producer = match &*producer {
                    Some(producer) => producer,
                    None => return false,
                };
```

(And even if you do get scheduled multiple times, the failure mode isn't too bad; you just double print the log message.)